legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/view/NavigatorImpl.kt

```diff
@@ -0,0 +1,23 @@
+package com.hamdan.forzenbook.legacy.core.view
+
+import android.content.Context
+
+class NavigatorImpl(
+    val legacyLoginContext: Context,
```

Comment:
Bring the context in the _methods_, not the constructor. Then you don't have to hold the context and be worried about leaks. Other than that your methods commented code looks about right

Reply:
updated